system-addon/lib/TopSitesFeed.jsm

```diff
@@ -284,10 +289,16 @@ this.TopSitesFeed = class TopSitesFeed {
    * Handle an insert (drop/add) action of a site.
    */
   insert(action) {
+    let {index} = action.data;
+    // Negative index will prepend the TopSite in the first position
+    if (index === -1) {
+      index = 0;
```

Comment:
You can skip the `|| 0` below with:
```js
// Treat invalid pin index values (e.g., -1, undefined) as insert in the first position
if (!(index > 0)) {
  index = 0;
}
```

(NB: The `>` implicitly converts `undefined` to `NaN` and `null` to `0`. And comparisons with `NaN` is `false`, so that's why the negation.)